src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3451,6 +3480,13 @@ private TypeSymbolWithAnnotations VisitOptionalImplicitConversion(BoundExpressio
 
             (BoundExpression operand, Conversion conversion) = RemoveConversion(expr, includeExplicitConversions: false);
             var operandType = VisitRvalueWithResult(operand);
+
+            var conv = expr as BoundConversion;
+            if (conv != null && conv.ConversionKind == ConversionKind.ImplicitNullable)
```

Comment:
Might there be a way to factor this logic into ApplyConversion?

Reply:
Perhaps although there are many callers of `ApplyConversion` but only two that need to track state of `Nullable<T>.Value`. Will consider moving to `ApplyConversion` when there are more scenarios.

---
In reply to: [250804661](https://github.com/dotnet/roslyn/pull/32729#discussion_r250804661) [](ancestors = 250804661)